### PR TITLE
[INLONG-8118][Manager] Support tenant user permission control

### LINKED ISSUE #8118: [Feature][Manager] Support tenant user permission control
### Description

parent #7914 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Fixes #8118 
- Parent #7914 

### Motivation

 Support tenant user permission control

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Getter
+    @JsonValue
+    private final Integer code;
+
+    InlongUserTypeEnum(Integer code) {
+        this.code = code;
+    }
+
+    private static final List<Integer> ARRAYS = Arrays.stream(values())
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java:47]: `ARRAYS` -> `TYPE_CODE_LIST`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java [hunk 2]
```diff
@@ -15,20 +15,24 @@
...
-    /**
-     * Get roles based on username
-     *
-     * @param username username
-     * @return Role list
-     */
-    List<String> listByUser(String username);
+public interface TenantRoleService {
+
+    PageInfo<TenantRoleInfo> listByCondition(TenantRolePageRequest request);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:31]: Please add Javadoc for those methods in interface class.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java (hunk 1 above)
Comment (vernedeng) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java:47]: fixed, thx

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java (hunk 2 above)
Comment (vernedeng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:34]: fixed, thx

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java (hunk 1 above)
Comment (healchow) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/InlongUserTypeEnum.java:47]: Not yet...

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    @OperationLog(operation = OperationType.CREATE)
+    @ApiOperation(value = "Save tenant role")
+    @RequiresRoles(value = UserRoleCode.ADMIN)
+    public Response<Integer> save(@Validated @RequestBody TenantRoleRequest request) {
+        return Response.success(tenantRoleService.save(request));
+    }
+
+    @RequestMapping(value = "/role/tenant/list", method = RequestMethod.POST)
+    @ApiOperation(value = "List tenant roles by paginating")
+    public Response<PageInfo<TenantRoleInfo>> listByCondition(@RequestBody TenantRolePageRequest request) {
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java:68]: @RequiresRoles(value = UserRoleCode.ADMIN)

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        Preconditions.expectNotBlank(tenantName, "Failed to save tenant user role, tenant should not be blank");
+        Preconditions.expectNotBlank(username, "Failed to save tenant user role, user should not be blank");
+        Preconditions.expectNotBlank(request.getRoleCode(),
+                "Failed to save tenant user role, role code should not be blank");
+
+        InlongTenantEntity tenant = tenantMapper.selectByName(tenantName);
+        Preconditions.expectNotNull(tenant, TENANT_NOT_EXIST, String.format(TENANT_NOT_EXIST.getMessage(), tenantName));
+
+        TenantUserRoleEntity entity = CommonBeanUtils.copyProperties(request, TenantUserRoleEntity::new);
+        String operator = LoginUserUtils.getLoginUser().getName();
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java:75]: fixed, thx

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java (hunk 3 above)
Comment (vernedeng) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java:68]: No need to require admin permission for query interfaces such as **_list_** and **_get_**

Comment (healchow) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongTenantRoleController.java:79]: Okay~

**Review by healchow — APPROVED** [review]

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantUserRoleEntity.java [hunk 5]
```diff
@@ -26,13 +26,13 @@
  * User role entity, including username, role code, etc.
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantUserRoleEntity.java:26]: Tenant user role entity

File: inlong-manager/manager-dao/src/main/resources/mappers/TenantUserRoleEntityMapper.xml [hunk 6]
```diff
@@ -33,32 +34,50 @@
...
+    <select id="listByCondition"
+            parameterType="org.apache.inlong.manager.pojo.user.TenantRolePageRequest"
+            resultMap="BaseResultMap">
+      select
         <include refid="Base_Column_List"/>
-        from user_role
-        where user_name = #{username,jdbcType=VARCHAR}
+        from tenant_user_role
+        where
+            is_deleted = 0
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/resources/mappers/TenantUserRoleEntityMapper.xml:61]: where  is_deleted = 0

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java [hunk 7]
```diff
@@ -15,20 +15,36 @@
...
 
-import java.util.List;
+import org.apache.inlong.manager.pojo.user.TenantRoleInfo;
+import org.apache.inlong.manager.pojo.user.TenantRolePageRequest;
+import org.apache.inlong.manager.pojo.user.TenantRoleRequest;
+
+import com.github.pagehelper.PageInfo;
 
 /**
  * Role service
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleService.java:27]: Tenant role service

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+import com.github.pagehelper.PageHelper;
+import com.github.pagehelper.PageInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import static org.apache.inlong.manager.common.enums.ErrorCodeEnum.TENANT_NOT_EXIST;
+
+/**
+ * Role operation
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java:43]: Tenant role operation

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantUserRoleEntity.java (hunk 5 above)
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantUserRoleEntity.java:26]: fixed, thx

File: inlong-manager/manager-dao/src/main/resources/mappers/TenantUserRoleEntityMapper.xml (hunk 6 above)
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/resources/mappers/TenantUserRoleEntityMapper.xml:61]: fixed, thx

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**